src/tlv/TlvNumber.ts

```diff
@@ -34,12 +34,11 @@ export class TlvNumericSchema<T extends bigint | number> extends TlvSchema<T> {
     override decodeTlvInternalValue(reader: TlvReader, typeLength: TlvTypeLength) {
         if (typeLength.type !== this.type) throw new Error(`Unexpected type ${typeLength.type}, was expecting ${this.type}.`);
         const value = reader.readPrimitive(typeLength) as T;
-        this.validate(value);
         return value;
     }
 
     override validate(value: T): void {
-        super.validate(value);
+        if (typeof value !== "number" && typeof value !== 'bigint') throw new Error(`Expected number, got ${typeof value}.`);
```

Comment:
Also validate in TlvNumberSchema to check that it is a number.

Reply:
Oh, I missed that second class in the file, added